src/pudl_rmi/connect_deprish_to_ferc1.py

```diff
@@ -66,762 +180,391 @@ def execute(plant_parts_eia, deprish_eia, ferc1_to_eia, clobber=False):
-
-        This method associates slices of ferc1 records - which are larger than
-        their depreciation counter parts - via prioritized columns.
-
-        Args:
-            same_smol (pandas.DataFrame): table with matched records from ferc1
-                and depreciation records.
-            data_col (string): name of the ferc1 data column.
-            idx_cols_ferc1 (list): columns to group by.
-            split_cols (list): ordered list of columns to split porportionally
-                based on. Ordered based on priority: if non-null result from
-                frist column, result will include first column result, then
-                second and so on.
-        Returns:
-            pandas.DataFrame: a modified version of `same_smol` with a new
-                assigned data_col
 
-        """
-        # we want to know the sum of the potential split_cols for each ferc1
-        # option
-        df_fgb = (
-            same_smol.loc[:, idx_cols_ferc1 + split_cols]
-            .groupby(by=idx_cols_ferc1, dropna=False)
-            .sum(min_count=1)
-            .reset_index()
-        )
-        df_w_tots = (
-            pd.merge(same_smol, df_fgb,
-                     on=idx_cols_ferc1,
-                     suffixes=("", "_fgb"))
-        )
-        # for each of the columns we want to split the frc data by
-        # generate the % of the total group, so we can split the data_col
-        new_data_col = self._get_clean_new_data_col(data_col)
-        df_w_tots[new_data_col] = pd.NA
-        for split_col in split_cols:
-            df_w_tots[f"{split_col}_pct"] = (
-                df_w_tots[split_col] / df_w_tots[f"{split_col}_fgb"])
-            # choose the first non-null option.
-            df_w_tots[new_data_col] = (
-                df_w_tots[new_data_col].fillna(
-                    df_w_tots[data_col] * df_w_tots[f"{split_col}_pct"]))
-
-        # merge in the newly generated split/assigned data column
-        df_final = pd.merge(
-            same_smol,
-            df_w_tots[idx_cols_ferc1 + ['record_id_eia_deprish',
-                                        new_data_col]].drop_duplicates(),
-        )
-        return df_final
+def str_squish(x):
+    """Squish strings from a groupby into a list."""
+    return '; '.join(list(map(str, [x for x in x.unique() if x is not pd.NA])))
 
-    def agg_ferc_data_cols(self):
-        """
-        Aggregate smaller level FERC1 data columns to depreciation records.
 
-        When the depreciation matches are at a higher level than the FERC1,
-        this method aggregates the many FERC1 records into the level of the
-        depreciation records.
-        """
-        data_cols_own_frac = [f"{col}_own_frac" for col in DATA_COLS_TO_SPLIT]
-        same_beeg = self.matches_df.loc[
-            (self.matches_df.match_method == 'same_quals')
-            & (self.matches_df.level_deprish == 'beeg')
-        ]
+def split_data_on_split_cols(
```

Comment:
I believe this function is.... allocating or distributing. Some of the language in here was kind of off because this is one of the functions that I effectively took from the old version. But let's chat about this and settle on some names that feel clear.